[ERROR] Failed to load theme settings. fetch failed

When trying to update the Helmholtz RSD to v2, the Helmholtz frontpage did not lod. The themes settings are not correctly loaded because the fetch fails. As a result, custom landing pages and themes are not loaded:

```typescript
// frontend/config/getSettingsSererSide.ts

export async function getRsdSettings() {
  try {
    // node request to localhost to load local json file
    const url = 'http://localhost:3000/data/settings.json'
    const resp = await fetch(url)
    if (resp.status === 200) {
      const json: RsdSettingsState = await resp.json()
      return json
    } else {
      logger(`Failed to load theme settings. ${resp.status} ${resp.statusText}`, 'warn')
      // return default settings
      return defaultSettings
    }
  } catch (e: any) {
    logger(`Failed to load theme settings. ${e.message}`, 'error')
    // return default settings
    return defaultSettings
  }
}
```

The error from the last catch block is:

```
TypeError: fetch failed
    at Object.fetch (node:internal/deps/undici/undici:11576:11)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async c (/app/.next/server/chunks/4496.js:1:10897)
    at async Promise.all (index 1)
    at async l (/app/.next/server/chunks/4496.js:1:11335)
    at async E.getInitialProps (/app/.next/server/chunks/4496.js:1:13641)
    at async s (/app/node_modules/next/dist/compiled/next-server/pages.runtime.prod.js:1:32006)
    at async Be (/app/node_modules/next/dist/compiled/next-server/pages.runtime.prod.js:1:53734)
    at async doRender (/app/node_modules/next/dist/server/base-server.js:1265:26)
    at async cacheEntry.responseCache.get.incrementalCache.incrementalCache (/app/node_modules/next/dist/server/base-server.js:1442:28) {
  cause: Error: connect ECONNREFUSED 127.0.0.1:3000
      at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1595:16) {
    errno: -111,
    code: 'ECONNREFUSED',
    syscall: 'connect',
    address: '127.0.0.1',
    port: 3000
  }
}
```

Upon closer look at the log of the RSD landing page with the current main branch, there are more errors:

```
rsd-frontend-1  | [ERROR] getPageLinks: fetch failed
rsd-frontend-1  | [ERROR] getAnnouncement: fetch failed
rsd-frontend-1  | [ERROR] Failed to load theme settings. fetch failed
rsd-frontend-1  | [ERROR] getHomepageCounts failed: fetch failed
```

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -108,7 +108,7 @@ services:
       # dockerfile to use for build
       dockerfile: Dockerfile
     # update version number to correspond to frontend/package.json
-    image: rsd/frontend:2.0.1
+    image: rsd/frontend:2.0.2
     environment:
       # it uses values from .env file
       - POSTGREST_URL

--- a/frontend/Dockerfile
+++ b/frontend/Dockerfile
@@ -24,7 +24,7 @@ RUN yarn install --frozen-lockfile --silent
 # ----------------------------------------
 # 2. Build image
 # ----------------------------------------
-FROM node:18.5-buster-slim AS builder
+FROM deps AS builder
 
 WORKDIR /app
 
@@ -59,18 +59,20 @@ ENV NODE_ENV production
 RUN addgroup --system --gid 1001 nodejs
 RUN adduser --system --uid 1001 nextjs
 
-# You only need to copy next.config.js if you are NOT using the default configuration
-# COPY --from=builder /app/next.config.js ./
-COPY --from=builder /app/public ./public
-COPY --from=builder /app/package.json ./package.json
-
 # Using standalone output to reduce image size
 # https://nextjs.org/docs/advanced-features/output-file-tracing
+COPY --from=builder /app/public ./public
 COPY --from=builder --chown=nextjs:nodejs /app/.next/standalone ./
 COPY --from=builder --chown=nextjs:nodejs /app/.next/static ./.next/static
 
 USER nextjs
 
 EXPOSE 3000
+
+# required to enable fetch to localhost:3000
+# set next port
+ENV PORT 3000
+# set next hostname
+ENV HOSTNAME "0.0.0.0"
 
 CMD ["node", "server.js"]

